### PR TITLE
Add miner_setGasLimit

### PR DESCRIPTION
**Pull request by timbeiko**
Format copied from miner_setGasPrice

### CODE REVIEW

File: docs/_rpc/ns-miner.md [hunk 1]
```diff
@@ -70,3 +70,13 @@ Sets the etherbase, where mining rewards will go.
 | Go      | `miner.SetEtherbase(common.Address) bool`                   |
 | Console | `miner.setEtherbase(address)`                               |
 | RPC     | `{"method": "miner_setEtherbase", "params": [address]}`     |
+
+### miner_setGasLimit
+
+Sets the gas limit the miner will target when minning.
```
Comment (MariusVanDerWijden) [inline docs/_rpc/ns-miner.md:76]:
```suggestion
Sets the gas limit the miner will target when mining. 
```

Comment (MariusVanDerWijden) [reply docs/_rpc/ns-miner.md:76]: We should document behaviour before and after london hf

Comment (timbeiko) [reply docs/_rpc/ns-miner.md]: @MariusVanDerWijden you mean the fact that it needs to be 2x'ed after London to get the same throughput? If so, I can add a line about that.

File: docs/_rpc/ns-miner.md [hunk 2]
```diff
@@ -70,3 +70,13 @@ Sets the etherbase, where mining rewards will go.
 | Go      | `miner.SetEtherbase(common.Address) bool`                   |
 | Console | `miner.setEtherbase(address)`                               |
 | RPC     | `{"method": "miner_setEtherbase", "params": [address]}`     |
+
+### miner_setGasLimit
+
+Sets the gas limit the miner will target when mining.
```
Comment (timbeiko) [inline docs/_rpc/ns-miner.md:76]:
```suggestion
Sets the gas limit the miner will target when mining. Note: on networks where EIP-1559 is activated, this should be set to twice what you want the gas target (i.e. the effective gas used on average per blocks) to be. 
```

Comment (timbeiko) [reply docs/_rpc/ns-miner.md:76]:
```suggestion
Sets the gas limit the miner will target when mining. Note: if EIP-1559 is activated, you should target 2x what you want the average block size to be. For example, a gas limit of 20 000 000 on a network with EIP-1559 is roughly equivalent to one of 10 000 000 on a network without it.
```

Alternative :-)

**Comment by MariusVanDerWijden** [comment]
Thank you for your contribution :stuck_out_tongue_winking_eye:

**Pull request merged**